commit 9f581d0cdf3423135be3283137fd1370f1d9303f
Author: Christoph Feck <christoph@maxiom.de>
Date:   Mon Oct 21 18:11:02 2013 +0200

    Fix crash with a single wallpaper "slideshow"
    
    To avoid the last image shown again, it was removed
    from the list of next images to show. If, however, it
    was the only image, no images were left.
    
    BUG: 323901
    FIXED-IN: 4.11.3
    REVIEW: 113368

diff --git a/plasma/generic/wallpapers/image/image.cpp b/plasma/generic/wallpapers/image/image.cpp
index 15e0e3f..317f668 100644
--- a/plasma/generic/wallpapers/image/image.cpp
+++ b/plasma/generic/wallpapers/image/image.cpp
@@ -768,6 +768,11 @@ void Image::nextSlide()
         // the last one picked from the previous set
         if (!previousPath.isEmpty()) {
             m_unseenSlideshowBackgrounds.removeAll(previousPath);
+
+            // prevent empty list
+            if (m_unseenSlideshowBackgrounds.isEmpty()) {
+                m_unseenSlideshowBackgrounds = m_slideshowBackgrounds;
+            }
         }
     }
 
